src/main/java/uk/gov/companieshouse/insolvency/delta/mapper/Utils.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.companieshouse.insolvency.delta.mapper;
+
+import static uk.gov.companieshouse.api.delta.CaseNumber.CaseTypeEnum.ADMINISTRATION;
+import static uk.gov.companieshouse.api.delta.CaseNumber.CaseTypeEnum.FOREIGN_INSOLVENCY;
+import static uk.gov.companieshouse.api.delta.CaseNumber.CaseTypeEnum.IN_ADMINISTRATION;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.ADMINISTRATION_DISCHARGED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.ADMINISTRATION_ENDED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.ADMINISTRATION_STARTED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.CONCLUDED_WINDING_UP_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.DECLARATION_SOLVENT_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.DISSOLVED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.DUE_TO_BE_DISSOLVED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.INSTRUMENTED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.MORATORIUM_STARTED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.PETITIONED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.VOLUNTARY_ARRANGEMENT_ENDED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.VOLUNTARY_ARRANGEMENT_STARTED_ON;
+import static uk.gov.companieshouse.api.insolvency.CaseDates.TypeEnum.WOUND_UP_ON;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import uk.gov.companieshouse.api.delta.CaseNumber;
+import uk.gov.companieshouse.api.insolvency.CaseDates;
+
+public final class Utils {
```

Comment:
MapperUtils probably better name for this class?

Reply:
Done